### ISSUE TITLE
Grave reverted to an earlier death.

### ISSUE THREAD

**Issue by Desim8r** [DarkPacks/SevTech-Ages#2728]
## Bug Report

Background:
Had previously died to the Hydra boss in the Twilight. Grave failed to spawn or was spawned and destroyed. Recovered inventory using /tmg_restore <player> latest.

Issue:
Died while self sacrificing using the blood altar in the overworld many game days later. Respawned at bed but no grave to be found where I died. Checked the grave marker and it showed a grave location in the Twilight. Short trip to the twilight and it was clear that it was point to the hydra den. No grave at that location. 

Ran /tmg_restore an the items recovered were the items from the previous death to the Hydra. There now appears to be no way to recover the items from my latest death.

## Expected Behaviour
I get my shit back :)

There are two issues at play here. Resolving the issue with Hydra and graves would probably avert the secondary bug however there is still the issue with Tomb Many Graves. 
For some reason the failure of the grave to spawn causes an issue with the list of graves the mod keeps, resulting in it losing track of the latest grave or failing to generate a new grave properly after that point.

It's a pretty severe bug as it seems your items are permanently lost.  

## Steps to Reproduce (for bugs)

See report above

## Logs

Server play. No access to log.

## Client Information
3.0.7
1.8.0_171
Twitch
8gb
Server
* Optifine Installed: No
* Shaders Enabled: No

## Server Information
* Java Version: Unknown
* Operating System: Linux
* Hoster/Hosting Solution: Creeperhost
* Sponge (Non-Vanilla Forge) Server: NA
* Additional Content Installed: NA

**Comment by artdude543** [comment]
Duplicate of #2429

**Comment by sampy901** [comment]
Actually, this is not a duplicate. I had this exact same issue just an hour ago. I had died to the Litch in the Litch Tower and picked up my grave several in-game and real-life days ago. Today I accidentally killed myself using the Blood Altar. My grave did not appear and when I did tmg_restore I got the items from when I died at the Litch Tower. It seems to have something to do with the blood altar and dying in the twilight forest. This is not a duplicate of 2429, which is a problem with shoggoth acid breaking graves.
